internal/providers/azure/azure.go

```diff
@@ -66,29 +64,52 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
-	if err != nil {
-		return types.Config{}, report.Report{}, err
+	for _, ovfFsType := range ovfFsTypes {
+		devices, err := execUtil.GetBlockDevices(ovfFsType)
+		if err != nil {
+			return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=%q: %v", ovfFsType, err)
+		}
+		for len(devices) == 0 {
+			// wait for the actual config drive to appear
+			// if it's not shown up yet
+			time.Sleep(time.Second)
+			devices, _ = execUtil.GetBlockDevices(ovfFsType)
+		}
+		for _, dev := range devices {
+			// verify that this is a CD-ROM drive with a UDF
```

Comment:
nit: s/with a UDF volume attached to it//, since Azure Stack also uses ISO 9660.